### DISCUSSION TITLE
Why doesn't my Custom error message work in IClaimsAuthoriser?

### DISCUSSION THREAD

**Discussion by sa-es-ir** [ThreeMammals/Ocelot#1777] [Q&A]
Hi, I want to use custom error when http status code was ``Forbidden(403)``, so for reach this I try this:

```C#
public class CustomClaimsAuthoriser : IClaimsAuthoriser
{
    public Response<bool> Authorise(
        ClaimsPrincipal claimsPrincipal,
        Dictionary<string, string> routeClaimsRequirement,
        List<PlaceholderNameAndValue> urlPathPlaceholderNameAndValues)
    {
        // some code
        return new ErrorResponse<bool>(new UserDoesNotHaveClaimError("User does not have required claims"));
    }
}
```
But response has no content and just return status code 403.
My code is wrong or what's the problem?
## Specifications

  - Ocelot Version: 16.0.1
  - Asp.net core 3.1

**Comment by ks1990cn** [comment]
What `content `you expect when Unauthorised?

**Comment by sa-es-ir** [comment]
@ks1990cn Pretty old issue and I don't remember how I handled that.
Because of better user experience on API response, I wanted to send a message along with a 403 status code.

**Reply by raman-m** [reply]
> IClaimsAuthoriser

We don't have this interface in Ocelot code base: [Search for IClaimsAuthoriser](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20IClaimsAuthoriser&type=code)
There is no such `IClaimsAuthoriser` interface in ASP.NET repository too: [Search for IClaimsAuthoriser in ASP.NET Core](https://github.com/search?q=repo%3Adotnet%2Faspnetcore%20IClaimsAuthoriser&type=code)

Cannot get it... Is it your custom interface? Are you kidding?

**Comment by raman-m** [comment]
> @sa-es-ir commented [Jan 15, 2021](#discussion-5813141)

Hi Saeed!
Please, upload your solution to your GitHub repo to demonstrate the bug... It is hard to guess and recommend something, if there's no real code to review.
You can fork the Ocelot repo and create feature branch with solution, or it could be separate repo where you make reference to Ocelot NuGet package.
Thanks!

Also, could you check these links to get an inspiration:
- Stack Overflow: [Ocelot RouteClaimsRequirement does not recognize my claims and returns 403 Forbidden](https://stackoverflow.com/questions/63049670/ocelot-routeclaimsrequirement-does-not-recognize-my-claims-and-returns-403-forbi)
